src/main/java/org/dataone/hashstore/filehashstore/FileHashStore.java

```diff
@@ -540,266 +508,66 @@ private ObjectMetadata syncPutObject(
-        InterruptedException {
-        FileHashStoreUtility.checkNotNegativeOrZero(objSize, "storeObject");
-
-        return storeObject(object, pid, null, null, null, objSize);
-    }
-
-    /**
-     * Overload method for storeObject with an additionalAlgorithm
-     */
-    @Override
-    public ObjectMetadata storeObject(InputStream object, String pid, String additionalAlgorithm)
-        throws NoSuchAlgorithmException, IOException, PidRefsFileExistsException, RuntimeException,
-        InterruptedException {
-        FileHashStoreUtility.ensureNotNull(
-            additionalAlgorithm, "additionalAlgorithm", "storeObject"
-        );
-
-        return storeObject(object, pid, additionalAlgorithm, null, null, -1);
-    }
-
-    @Override
-    public boolean verifyObject(
-        ObjectMetadata objectInfo, String checksum, String checksumAlgorithm, long objSize
-    ) throws IOException, NoSuchAlgorithmException, IllegalArgumentException {
-        logFileHashStore.debug(
-            "FileHashStore.verifyObject - Called to verify object with id: " + objectInfo.getCid()
-        );
-        FileHashStoreUtility.ensureNotNull(objectInfo, "objectInfo", "verifyObject");
-        FileHashStoreUtility.ensureNotNull(checksum, "checksum", "verifyObject");
-        FileHashStoreUtility.ensureNotNull(checksumAlgorithm, "checksumAlgorithm", "verifyObject");
-        FileHashStoreUtility.checkNotNegativeOrZero(objSize, "verifyObject");
-
-        Map<String, String> hexDigests = objectInfo.getHexDigests();
-        String digestFromHexDigests = hexDigests.get(checksumAlgorithm);
-        long objInfoRetrievedSize = objectInfo.getSize();
-        String objCid = objectInfo.getCid();
-
-        if (objInfoRetrievedSize != objSize) {
-            logFileHashStore.info(
-                "FileHashStore.verifyObject - Object size invalid for cid: " + objCid
-                    + ". Expected size: " + objSize + ". Actual size: " + objInfoRetrievedSize
-            );
-            return false;
-
-        } else if (!digestFromHexDigests.equals(checksum)) {
-            logFileHashStore.info(
-                "FileHashStore.verifyObject - Object content invalid for cid: " + objCid
-                    + ". Expected checksum: " + checksum + ". Actual checksum calculated: "
-                    + digestFromHexDigests + " (algorithm: " + checksumAlgorithm + ")"
-            );
-            return false;
-
-        } else {
-            logFileHashStore.info(
-                "FileHashStore.verifyObject - Object has been validated for cid: " + objCid
-                    + ". Expected checksum: " + checksum + ". Actual checksum calculated: "
-                    + digestFromHexDigests + " (algorithm: " + checksumAlgorithm + ")"
-            );
-            return true;
-        }
-    }
-
     @Override
     public void tagObject(String pid, String cid) throws IOException, PidRefsFileExistsException,
```

Comment:
This method can be used by client directly. So do you think we need to create another lock for pids?

Reply:
Thank you for catching this! Yes we should and I have added in a new lock. I have also revised the existing locks so that they are more clear in terms of what is actually being locked.